lib/ansible/plugins/inventory/vmware_vm_inventory.py

```diff
@@ -31,12 +31,12 @@
             description: Name of vCenter or ESXi server.
             required: True
             env:
-              - name: VMWARE_SERVER
+              - name: VMWARE_HOST
         username:
```

Comment:
I believe change has to be done in injector.py file instead of here.
What if other clients are using this env variable? I did search for VMWARE_USERNAME, found these references https://github.com/ansible/ansible/search?q=VMWARE_USERNAME&unscoped_q=VMWARE_USERNAME

Reply:
@AlanCoding @wenottingham looking at `job_env` on an inventory update I see 
```
        "VMWARE_USER": "administrator@vsphere.local",
        "VMWARE_PASSWORD": "**********",
        "VMWARE_HOST": "65.215.200.2",
        "VMWARE_VALIDATE_CERTS": "False",
```
beyond that I'm not sure how Tower is handling the default VMware credential and where a change would be needed

Reply:
AWX does not yet use this inventory plugin. When we do, obviously we will want to use the most future-proof parameters. For current vmware inventory imports we use the script, and its config follows this pattern:

https://github.com/ansible/ansible/blob/devel/contrib/inventory/vmware.ini

You can see we put the host/username/password in that config file

https://github.com/ansible/awx/blob/4a1c1217922570dc8361d33efd3b14d74529e645/awx/main/models/inventory.py#L2455

This will be going away entirely as the plugin replaces it.

However, this is how credentials are applied for _job_ (playbook) runs:

https://github.com/ansible/awx/blob/4a1c1217922570dc8361d33efd3b14d74529e645/awx/main/models/credential/injectors.py#L68-L72

That uses `VMWARE_HOST` that you mention.

AWX _can_ use different environment variables for inventory updates versus playbook runs. But why should we? Is it not reasonable to expect that inventory plugins accept credentials by the same environment variables as the modules?

So to answer what we _should_ do, we should look to the vmware modules for example.

https://docs.ansible.com/ansible/2.6/modules/vmware_guest_module.html#vmware-guest-module

> If the value is not specified in the task, the value of environment variable VMWARE_HOST will be used instead.

This inventory script used VMWARE_HOST

https://github.com/ansible/ansible/blob/devel/contrib/inventory/vmware.py

But _this_ inventory script (the one used by AWX) uses VMWARE_SERVER

https://github.com/ansible/ansible/blob/devel/contrib/inventory/vmware_inventory.py

There might be a reason for this. Maybe they server different purposes, and the server vs. host refers to different things. But maybe not, I don't know.

Reply:
Added `VMWARE_HOST` as environment variable to match value used by tower and vmware_* modules